web/modules/custom/dpl_opening_hours/src/Mapping/RepetitionMapper.php

```diff
@@ -0,0 +1,63 @@
+<?php
+
+namespace Drupal\dpl_opening_hours\Mapping;
+
+use DanskernesDigitaleBibliotek\CMS\Api\Model\DplOpeningHoursCreatePOSTRequestRepetition as RequestRepetition;
+use DanskernesDigitaleBibliotek\CMS\Api\Model\DplOpeningHoursListGET200ResponseInnerRepetition as ResponseRepetition;
+use DanskernesDigitaleBibliotek\CMS\Api\Model\DplOpeningHoursListGET200ResponseInnerRepetitionWeeklyData as WeeklyData;
+use Drupal\dpl_opening_hours\Model\Repetition\NoRepetition;
+use Drupal\dpl_opening_hours\Model\Repetition\Repetition;
+use Drupal\dpl_opening_hours\Model\Repetition\WeeklyRepetition;
+
+/**
+ * Mapper between value objects and OpenAPI request/response objects.
+ */
+class RepetitionMapper {
+
+  /**
+   * Map a request repetition to a value object.
+   */
+  public function fromRequest(RequestRepetition|ResponseRepetition $data): Repetition {
+    $repetitionType = ($data->getType() !== NULL) ? OpeningHoursRepetitionType::tryFrom($data->getType()) : NULL;
```

Comment:
Cant you just do it like 
```suggestion
    $repetitionType =OpeningHoursRepetitionType::tryFrom($data?->getType());
```

to make it a bit simpler?

Reply:
Fair enough, although, couldnt you just mark the string parameter as  `?string`? i get it's a bit silly to pass null to something, as it will never be a valid value, but it makes stuff like this much simpler

Reply:
For the record: `tryFrom()` is not a function we have defined. It is from the PHP Enum handling. As such we cannot change it.